### ISSUE TITLE
[BUG] Unable to compile after fresh installation 

### ISSUE THREAD

**Issue by yashug** [petyosi/react-virtuoso#832]
**Describe the bug**
./node_modules/react-virtuoso/dist/index.mjs 470:64
Module parse failed: Unexpected token (470:64)
File was processed with these loaders:
 * ./node_modules/babel-loader/lib/index.js
You may need an additional loader to handle the result of these loaders.
|   const log = statefulStream(function (label, message) {
|     let level = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : 1;
>     const currentLevel = getGlobalThis()["VIRTUOSO_LOG_LEVEL"] ?? getValue(logLevel);
|     if (level >= currentLevel) {
|       console[CONSOLE_METHOD_MAP[level]]("%creact-virtuoso: %c%s %o", "color: #0253b3; font-weight: bold", "color: initial", label, message);

**Comment by petyosi** [comment]
Reproduce this in a sample project for me to be able to help.

**Comment by gguiceuna** [comment]
I'm also seeing this after upgrading from 2.16.6 to 4.1.0. I tried every version of react-virtuoso 4.x.x and it started working again when instead used 3.1.5.

**Comment by adirzoari** [comment]
@yashug same error. did you find any solution?

**Comment by yashug** [comment]
@adirzoari no. I have moved to react window library instead

**Comment by adirzoari** [comment]
@yashug you also try to do it with typescript?

**Comment by yashug** [comment]
@adirzoari yes

**Comment by adirzoari** [comment]
@petyosi actually I took the same example that material-ui provide [https://codesandbox.io/s/piqe37?file=/demo.tsx](https://codesandbox.io/s/piqe37?file=/demo.tsx) and I got the same error. I'm using typescript, maybe need to add any configuration? any @types package?

**Comment by petyosi** [comment]
There's no types package. The types are shipped in the same package. 

My guess is that there's something to do with your setup not understanding the newish ecmascript syntax in the code. 

Again, I need a reproduction to help further. The code sandbox you refer to works well by itself as far as I can see.

**Comment by adirzoari** [comment]
@petyosi actually I can't add my whole project into sandbox.. is the setup include more steps or just install the package?

**Comment by petyosi** [comment]
You can add the relevant parts (bundling) into a GitHub repository. As long as it is runnable and shows the problem, it will help me help you.

**Comment by adirzoari** [comment]
@petyosi actually I added here sample of the code: [https://codesandbox.io/s/modern-flower-e8j01x?file=/src/index.tsx](https://codesandbox.io/s/modern-flower-e8j01x?file=/src/index.tsx) 
but here it's working but in my computer (node v16.8.0, npm 7.21.0) I'm getting the error
```
`./node_modules/react-virtuoso/dist/index.mjs 469:64
Module parse failed: Unexpected token (469:64)
File was processed with these loaders:
 * ./node_modules/babel-loader/lib/index.js
You may need an additional loader to handle the result of these loaders.
|   const log = statefulStream(function (label, message) {
|     let level = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : 1;
>     const currentLevel = getGlobalThis()["VIRTUOSO_LOG_LEVEL"] ?? getValue(logLevel);
|     if (level >= currentLevel) {
|       console[CONSOLE_METHOD_MAP[level]]("%creact-virtuoso: %c%s %o", "color: #0253b3; font-weight: bold", "color: initial", label, message);
```

**Comment by petyosi** [comment]
I can't be certain about the fix in your setup, but this looks like it: https://stackoverflow.com/questions/67638905/nullish-coalescing-operator-webpack-bundler-issues

**Comment by adirzoari** [comment]
@petyosi  umm not sure it relate to the issue.. I don't have babel config in my project

**Comment by ehoops-cz** [comment]
nullish coalescing operator is in `preset-env` now, so this shouldn't be the exact issue: https://babeljs.io/docs/babel-plugin-proposal-nullish-coalescing-operator.  I tried adding the `babel-plugin-transform-globalthis` polyfill, but I'm still getting the same error.  Our babel config is in my reply here: https://stackoverflow.com/questions/75478898/react-virtuoso-unable-to-compile-after-fresh-installation/75701439#75701439

**Comment by petyosi** [comment]
While researching something else, I reproduced that error. In that particular case, the problem was caused by `react-scripts` v4 - I can't say for sure if this is what causes it in the setup above. 

The problem should be fixed in react-virtuoso@v4.2.1.
